File: best-time-to-buy-and-sell-stock/maxProfit.go
func maxProfit(prices []int) int {
    var left, right, profit, maxProfit int

    for right < len(prices) {
        if(prices[right] > prices[left]) {
            profit = prices[right] - prices[left]
            maxProfit = max(profit, maxProfit)
        } else {
            left = right
        }
        
        right++
    }

    return maxProfit
}

func max(a int, b int) int {
    if(a > b) {
        return a
    }
    return b
}

File: best-time-to-buy-and-sell-stock/maxProfit.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function (prices) {
  let left = 0;
  let right = 0;
  let maxProfit = 0;
  while (prices.length > right) {
    if (prices[right] > prices[left]) {
      let profit = prices[right] - prices[left];
      maxProfit = Math.max(maxProfit, profit);
    } else {
      left = right;
    }
    right++;
  }

  return maxProfit;
};
